libass/ass_shaper.c

```diff
@@ -1004,17 +1025,23 @@ FriBidiStrIndex *ass_shaper_reorder(ASS_Shaper *shaper, TextInfo *text_info)
     for (i = 0; i < text_info->length; i++)
         shaper->cmap[i] = i;
 
-    // Create reorder map line-by-line
-    for (i = 0; i < text_info->n_lines; i++) {
-        LineInfo *line = text_info->lines + i;
-        FriBidiParType dir = FRIBIDI_PAR_ON;
+    // Create reorder map line-by-line or run-by-run
+    int last_break = 0;
+    GlyphInfo *glyphs = text_info->glyphs;
+    for (i = 0; i < text_info->length; i++) {
```

Comment:
While at it, you can move `i` and `ret` definitions from the start of the function to the first use.